lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -68,10 +68,24 @@ class AwsCompileApigEvents {
           .then(this.compileStage);
       },
 
+      // TODO should be removed once AWS fixes the CloudFormation problems using a separate Stage
+      'after:deploy:deploy': () => {
+        const updateStage = require('./lib/hack/updateStage').updateStage; // eslint-disable-line
+
+        this.validated = this.validate();
+
+        if (this.validated.events.length === 0) {
+          return BbPromise.resolve();
+        }
+
+        return BbPromise.bind(this)
+          .then(updateStage);
```

Comment:
Assuming we plan to convert to native promises, it'll make it easier if we write it as:

```javascript
return new BpPromise(resolve => resolve(updateStage.call(this)));
```

Also if `updateStage` is guaranteed to be async, it'll be even better to write it just as:

```javascript
return updateStage.call(this);
```

Additional question is why in `updateStage`, we pass `serverless` as a context (`this`), wouldn't it be cleaner to just pass it as an argument?

Reply:
Sounds like a good plan. Will update that :+1:

Binding this is a pattern we've used throughout the whole codebase that why I decided to sick with this for now.